api/internal/target/kusttarget_test.go

```diff
@@ -258,6 +258,110 @@ metadata:
 	assert.Equal(t, string(expYaml), string(actYaml))
 }
 
+func TestMergeTransformersConfig(t *testing.T) {
+	th := kusttest_test.MakeHarness(t)
+	th.WriteK("/merge-config", `
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+namePrefix: foo-
+nameSuffix: -bar
+namespace: ns1
```

Comment:
Is there a reason we need `namespace` here?

I can understand that you'd want to include an additional field controlled by `configurations`, say, to test that your provided `configurations` don't affect this additional field. `nameSuffix` serves this purpose well because it deals with the same `path` as `namePrefix` (brilliant!). 

But then what is the role of `namespace`? If it doesn't serve a pointed additional purpose, I'd recommend removing it to make the test more readable.

Reply:
The initial idea is to test that the provided `configurations` does not affect `namespace`, however since I am already using `nameSuffix` for this, I agree that `namespace` can be removed. I already removed the namespace resource to keep the test concise.